### ISSUE TITLE
Let the `LearningRateMonitor` log the actual parameter group names, if they exist

### ISSUE THREAD

**Issue by AffineParameter** [PyTorchLightning/pytorch-lightning#7981]
## 🚀 Feature

As a `PytorchLightning` user, I would like to monitor parameter group learning rates that have been logged under the parameter group's given name.

### Motivation

My use case has _many_ parameter groups associated with different heads/tasks, each of which is partitioned within a unique (and static) parameter group. When using the `LearningRateMonitor` all these learning rates end up taking the form: `[pg1, pg2, ..., pgn]` ... making it hard/impossible to keep track of which `pg` maps to which task. See the attached TB screen shot below, to see what I mean.

### Pitch

Instead of `[pg1, pg2, ..., pgn]`, I want to take advantage of the existing parameter group naming functionality in `pytorch` like so:

```python
    class TestModel(BoringModel):

        def configure_optimizers(self):
            optimizer = torch.optim.SGD(
                [
                    {'params': [p for p in self.layer.parameters()], 'name': 'linear'}
                ],
                 lr=0.1
            )
            lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
            return [optimizer], [lr_scheduler]
```

### Alternatives

Alternatively, I write down in my log book what heads `[pg1, pg2, ..., pgn]` map to. 

### Additional context

![Screen Shot 2021-06-14 at 6 15 30 PM](https://user-images.githubusercontent.com/10703027/121970630-8d763c00-cd3c-11eb-8425-1ec2f156e1bc.png)

**Comment by AffineParameter** [comment]
I have what I believe to be a working branch ready to go, pending appropriate discussion. Hope y'all have a great day.

**Comment by tchaton** [comment]
Hey @AffineParameter,

Sounds interesting. Feel free to open a PR.

Best,
T.C

**Comment by AffineParameter** [comment]
Should be good for CR @tchaton, thanks for the go-ahead.

Cheers,
~A
